lib/services/local/headlessauthn_watcher.go

```diff
@@ -181,7 +182,6 @@ func (h *HeadlessAuthenticationWatcher) watch(ctx context.Context) error {
 
 	// Notify any waiters initiated before the new watcher initialized.
 	h.notify(headlessAuthns...)
-	h.retry.Reset()
```

Comment:
We don't need the reset anymore? I'm not super familiar with this, so I'm not sure I follow.